bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsNodeSyncImpl.java

```diff
@@ -0,0 +1,609 @@
+                        continue;
+                    } else {
+                        setProperty(value, key, parentNode);
+                    }
+                } catch (RepositoryException re) {
+                    LOG.warn("Repository exception thrown. Skipping '{}' single property for node '{}'.", key, parentNode.getPath());
+                }
+            }
+        }
+    }
+
+    /**
+     * Set mixins property for the property's node.
+     * @param json JSONObject
+     * @param key String
+     * @param node Node
+     * @throws JSONException exception
+     * @throws RepositoryException exception
+     */
+    private void setMixinsProperty(final JSONObject json, final String key, final Node node) throws JSONException, RepositoryException {
+        JSONArray mixins = (JSONArray) json.get(key);
+        for (int i = 0; i < mixins.length(); i++) {
+            node.addMixin(mixins.getString(i));
+        }
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Mixins added for node '{}'.", node.getPath());
+        }
+    }
+
+    /**
+     * Set tags property for the property's node.
+     * @param json JSONObject
+     * @param key String
+     * @param node Node
+     * @throws JSONException exception
+     * @throws RepositoryException exception
+     */
+    private void setTagsProperty(final JSONObject json, final String key, final Node node) throws JSONException, RepositoryException {
+        TagManager tagManager = this.resourceResolver.adaptTo(TagManager.class);
+        JSONArray tags = (JSONArray) json.get(key);
+        ArrayList<Tag> tagList = new ArrayList<>();
+
+        for (int i = 0; i < tags.length(); i++) {
+            Tag tag = tagManager.resolve(tags.getString(i));
+            if (tag == null) {
+                LOG.warn("Tag '{}' could not be found. Skipping tag for node '{}'.", tags.getString(i), node.getPath());
+                continue;
+            }
+
+            tagList.add(tag);
+        }
+
+        if (tagList.size() > 0) {
+            Resource parentResource = this.resourceResolver.getResource(node.getPath());
+            tagManager.setTags(parentResource, tagList.toArray(new Tag[0]));
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Tags added for node '{}'.", node.getPath());
+            }
+        }
+    }
+
+    /**
+     * Set generic array property for the property's node.
+     * @param json JSONObject
+     * @param key String
+     * @param node Node
+     * @throws JSONException exception
+     * @throws RepositoryException exception
+     */
+    private void setArrayProperty(final JSONObject json, final String key, final Node node) throws JSONException, RepositoryException {
+        JSONArray rawValues = (JSONArray) json.get(key);
+        Object firstVal = rawValues.get(0);
+        Value[] propertyValues = new Value[rawValues.length()];
+
+        if (firstVal instanceof Boolean) {
```

Comment:
I feel like a try/catch here might be good in case firstVal ends up being a different type than a latter value int he array.  For instance, will a property of [6, 3849830489034803, 8] see the first value as Integer and a latter value as Long (which could blow up trying to cast to Integer)?